apm-lambda-extension/apmproxy/metadata.go

```diff
@@ -39,18 +37,25 @@ func ProcessMetadata(data AgentData) ([]byte, error) {
 	if err != nil {
 		return nil, fmt.Errorf("Error uncompressing agent data for metadata extraction: %w", err)
 	}
-	scanner := bufio.NewScanner(strings.NewReader(string(uncompressedData)))
-	scanner.Scan()
-	if strings.Contains(strings.ToLower(scanner.Text()), "metadata") {
-		return scanner.Bytes(), nil
+
+	before, _, _ := bytes.Cut(uncompressedData, []byte("\n"))
+
+	// Try without lowercase first to avoid allocations.
+	if bytes.Contains(before, []byte("metadata")) {
+		return before, nil
 	}
+
+	if bytes.Contains(bytes.ToLower(before), []byte("metadata")) {
+		return before, nil
+	}
```

Comment:
Maybe we could even get rid of this check? Agents are expected to send a valid `{"metadata":...` line. If they don't, then the agent should be fixed; it's not the Lambda extension's responsibility to filter out junk. WDYT?